Gems/Atom/Feature/Common/Code/Source/PostProcessing/FilmGrainPass.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PostProcessing/FilmGrainPass.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Scene.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<FilmGrainPass> FilmGrainPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<FilmGrainPass> pass = aznew FilmGrainPass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        FilmGrainPass::FilmGrainPass(const RPI::PassDescriptor& descriptor)
+            : RPI::ComputePass(descriptor)
+        {
+        }
+
+        bool FilmGrainPass::IsEnabled() const
+        {
+            if (!ComputePass::IsEnabled())
+            {
+                return false;
+            }
+            const RPI::Scene* scene = GetScene();
+            if (!scene)
+            {
+                return false;
+            }
+            PostProcessFeatureProcessor* fp = scene->GetFeatureProcessor<PostProcessFeatureProcessor>();
+            const RPI::ViewPtr view = GetRenderPipeline()->GetDefaultView();
+            if (!fp)
+            {
+                return false;
+            }
+            PostProcessSettings* postProcessSettings = fp->GetLevelSettingsFromView(view);
+            if (!postProcessSettings)
+            {
+                return false;
+            }
+            const FilmGrainSettings* filmGrainSettings = postProcessSettings->GetFilmGrainSettings();
+            if (!filmGrainSettings)
+            {
+                return false;
+            }
+            return filmGrainSettings->GetEnabled();
+        }
+
+        void FilmGrainPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            // Must match the struct in FilmGrain.azsl
+            struct Constants
+            {
+                AZStd::array<u32, 2> m_outputSize;
+                AZStd::array<u32, 2> m_grainTextureSize;
+                float m_intensity = FilmGrain::DefaultIntensity;
+                float m_luminanceDampening = FilmGrain::DefaultLuminanceDampening;
+                float m_tilingScale = FilmGrain::DefaultTilingScale;
+            } constants{};
```

Comment:
Add a **float m_pad** at the end to ensure that the struct ends at 16 byte boundary